tests/foreman/api/test_http_proxy.py

```diff
@@ -125,3 +126,75 @@ def test_positive_end_to_end(setup_http_proxy, module_target_sat, module_manifes
     )
     assert len(yum_repo['output']) >= 1
     assert 'quay/busybox' in yum_repo['output']
+
+
+@pytest.mark.upgrade
+@pytest.mark.rhel_ver_match('8')
+@pytest.mark.run_in_one_thread
+@pytest.mark.parametrize(
+    'setup_http_proxy',
+    [None, True, False],
+    indirect=True,
+    ids=['no_http_proxy', 'auth_http_proxy', 'unauth_http_proxy'],
+)
+@pytest.mark.tier3
+def test_positive_auto_attach(
+    setup_http_proxy, module_target_sat, rhel_contenthost, function_entitlement_manifest_org
+):
+    """Attempt to auto attach a subscription to content host
+
+    :id: cce888b5-e023-4ee2-bffe-efa9260224ee
+
+    :customerscenario: true
+
+    :expectedresults: host successfully subscribed, subscription
+        repository enabled, and repository package installed.
+
+    :Assignee: jpathan
+
+    :BZ: 2046337
+
+    :parametrized: yes
+
+    :CaseLevel: System
+    """
+    org = function_entitlement_manifest_org
+    lce = module_target_sat.api.LifecycleEnvironment(organization=org).create()
+    content_view = module_target_sat.api.ContentView(organization=org).create()
+    rh_repo_id = enable_sync_redhat_repo(constants.REPOS['rhel8_bos'], org.id)
+    rh_repo = module_target_sat.api.Repository(id=rh_repo_id).read()
+    assert rh_repo.content_counts['rpm'] >= 1
+    content_view = module_target_sat.api.ContentView(
+        id=content_view.id, repository=[rh_repo]
+    ).update(["repository"])
+    content_view.publish()
+    content_view = content_view.read()
+    content_view.version[-1].promote(data={'environment_ids': lce.id})
+    subscription = module_target_sat.api.Subscription(organization=org.id).search(
+        query={'search': f'name="{constants.DEFAULT_SUBSCRIPTION_NAME}"'}
+    )
+    assert len(subscription)
+    activation_key = module_target_sat.api.ActivationKey(
+        content_view=content_view,
+        organization=org,
+        environment=lce,
+        auto_attach=False,
+    ).create()
+    activation_key.add_subscriptions(data={'subscription_id': subscription[0].id})
+    rhel_contenthost.install_katello_ca(module_target_sat)
+    rhel_contenthost.register_contenthost(org=org.name, activation_key=activation_key.name)
+    assert rhel_contenthost.subscribed
+    module_target_sat.cli.Host.subscription_register(
+        {
+            'organization-id': org.id,
+            'content-view-id': content_view.id,
+            'lifecycle-environment-id': lce.id,
+            'name': rhel_contenthost.hostname,
+        }
+    )
+    host = module_target_sat.api.Host().search(
+        query={'search': f'name={rhel_contenthost.hostname}'}
+    )[0]
+    # To Do: Use api
+    module_target_sat.cli.Host.subscription_auto_attach({'host-id': host.id})
+    rhel_contenthost.execute('yum install -y zsh')
```

Comment:
What are we trying to achieve with this line of code? `rhel_contenthost.execute('yum install -y zsh')`

Reply:
@jameerpathan111 What I would like to see is assert that the package was installed. Either by checking `rc` of `execute` or running `rpm -q`.

Reply:
@adarshdubey-star yes, the package will be available from the subscription and we are checking if we can install it when HTTP proxy is configured on satellite.

Reply:
@ogajduse added assertion.